wrappers/hnygingonic/gingonic.go

```diff
@@ -0,0 +1,82 @@
+package hnygingonic
+
+import (
+	"context"
+
+	"github.com/gin-gonic/gin"
+	"github.com/honeycombio/beeline-go"
+	"github.com/honeycombio/beeline-go/trace"
+	"github.com/honeycombio/beeline-go/wrappers/common"
+)
+
+const ginContextKey = "beeline-middleware-context"
+
+// Middleware wraps httprouter handlers. Since it wraps handlers with explicit
+// parameters, it can add those values to the event it generates.
+func Middleware(queryParams map[string]struct{}) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		// get a new context with our trace from the request, and add common fields
+		ctx, span := common.StartSpanOrTraceFromHTTP(c.Request)
+		defer span.Send()
+		// Add the span context to the gin context as we need to be able to pass
+		// this context around our gin application
+		c.Set(ginContextKey, ctx)
+		// push the context with our trace and span on to the request
+		c.Request = c.Request.WithContext(ctx)
+
+		// replace the writer with our wrapper to catch the status code
+		wrappedWriter := common.NewResponseWriter(c.Writer)
+
+		// pull out any variables in the URL, add the thing we're matching, etc.
+		for _, param := range c.Params {
+			span.AddField("handler.vars."+param.Key, param.Value)
+		}
```

Comment:
Does it make sense for this (and below in `Middleware`) to be `AddField` or would it be better for it to be `AddTraceField`?

Reply:
I can see the lure. My instinct is to leave trace level fields for the application developers to choose what gets threaded through all spans rather than having middleware do it (easy enough to add the same field within the handler) but I don't have compelling experience pushing me one way or the other.

Reply:
Understood, it also occurs to me that it could be seen as unexpected behaviour if the rest of the middleware packages are not doing it :+1: